src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -610,18 +611,16 @@ def set_ag_waf_config_2017_03_01(resource_group_name, application_gateway_name,
 
         # for disabled rules, we have to look up the IDs
         if disabled_rules:
-            results = list_ag_waf_rule_sets(
-                ncf, _type=rule_set_type, version=rule_set_version, group='*')
-            for item in results:
-                for group in item.rule_groups:
-                    disabled_group = ApplicationGatewayFirewallDisabledRuleGroup(
-                        group.rule_group_name, [])
-
-                    for rule in group.rules:
-                        if str(rule.rule_id) in disabled_rules:
-                            disabled_group.rules.append(rule.rule_id)
-                    if disabled_group.rules:
-                        disabled_groups.append(disabled_group)
+            results = list_ag_waf_rule_sets(ncf, _type=rule_set_type, version=rule_set_version, group='*')
```

Comment:
Or, alternatively, apply the same logic you do elsewhere and turn the block into a method and then running that method on each item in results.  That I would be okay with because it reduces the indenting and adds clarity.